include/feature_lib.h

```diff
@@ -0,0 +1,62 @@
+#ifndef FEATUREDETECT_FEATURELIB_H_
+#define FEATUREDETECT_FEATURELIB_H_
+
+#include <vector>
+#include <opencv2/features2d.hpp>
+
+namespace featurelib {
+
+struct KeyPoint {
+  float x;
+  float y;
+  float size;
+  float angle;
+  float strength;
+};
+
+struct DetectionResult {
+  std::vector<KeyPoint> keypoints;
+  std::vector<std::vector<uint8_t>> descriptors;
+};
+
+class FeatureDetector {
+ public:
+  virtual DetectionResult detect(const std::vector<std::uint8_t> &input_vector) const = 0;
+
+  void setHeight(int height) {
+    mask.reshape(1, std::vector<int>{height, width});
+  }
+
+  void setWidth(int width) {
+    mask.reshape(1, std::vector<int>{height, width});
+  }
+
+  int getWidth() const {
+    return width;
+  }
+
+  int getHeight() const {
+    return height;
+  }
+
+ protected:
+  int width;
+  int height;
```

Comment:
You need to initialize them to something, or they'll stay uninitialized in case of using the default constructor. CLion gives you a warning -- do you even read them?

Reply:
Yep, this is because the names are the same as those in the setters
even i initialize them to something clion is still giving me warnings

Reply:
maybe height_input and width_input ass parameters and also i initialize protected fields&

Reply:
Yep, the warning I'm talking about appears in the default constructor's
 implementation.

Let the names be just `value` in the setters: `void setWidth(int value)`